XMonad/Util/Loggers.hs

```diff
@@ -182,6 +186,72 @@ logLayout = withWindowSet $ return . Just . ld
 logCurrent :: Logger
 logCurrent = withWindowSet $ return . Just . W.currentTag
 
+-- | Log the given string, as is.
+logConst :: String -> Logger
+logConst = return . Just
+
+-- | Like they 'maybe' combinator, but for  loggers. If the given logger
+-- gives @Nothing@, the default logger is returned.
+-- For example, to display a quote when no windows are on the screen,
+-- you can do:
+--
+-- > logMaybe logTitle (logConst "Hey, you, you're finally awake.")
+logMaybe :: Logger -> Logger -> Logger
+logMaybe l d = l >>= maybe d logConst
+
+-- | An infix operator for 'logMaybe', which can be more convenient to
+-- combine multiple loggers.
+--
+-- > logTitle .| logWhenActive 0 (logConst "*") .| logConst "There's nothing here"
+(.|) :: Logger -> Logger -> Logger
+(.|) = logMaybe
+
+-- $xmonad-screen
+-- It is also possible to bind loggers like 'logTitle' to a specific screen. For
+-- example, using @logTitleOnScreen 1@ will log the title of the focused window
+-- on screen 1, even if screen 1 is not currently active.
+
+-- | Only display the 'Logger' if the screen with the given 'ScreenId' is
+-- active.
+-- For example, this can be used to create a marker for the active
+-- screen:
+--
+-- > logWhenActive 0 (logConst "*")
```

Comment:
I don't understand the example. :-(
Screen 0 is not necessarily the active one, is it?
I thought it's the first screen reported by Xinerama, so either the one on primary output or the first output.

Reply:
The comment is confusing. Basically, if screen 0 (the primary one) is active, the logger will return a "*".  I'll update it